### PR TITLE
Update sh_titan_embark.gnut to have the arm badge as a client side riff for FP embarks

### PR DESCRIPTION
**Pull request by Gityeeter**
Makes the Titan arm badge as a client side riff for toggling FP embarks.

### CODE REVIEW

File: Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut [hunk 1]
```diff
@@ -887,6 +887,7 @@ function TitanEmbark_PlayerEmbarks( entity player, entity titan, table e )
 	sequence.blendTime = 0.5
 
 	entity soul = titan.GetTitanSoul()
+
```
Comment (uniboi) [inline Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut:890]: funny whitespaces in the commit

File: Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut [hunk 2]
```diff
@@ -1087,9 +1105,19 @@ void function PhaseEmbarkPhaseStart( entity player )
 	player.MakeInvisible()
 	PlayPhaseShiftDisappearFX( player )
 	EmitSoundOnEntity( player, "pilot_phaseembark_activate_3p" )
-	
+
 	if ( GetCurrentPlaylistVarInt( "fp_embark_enabled", 0 ) == 1 )
+	{
 		player.PhaseShiftBegin( 0.0, 0.2 )
+	}
+	else if((player.GetPetTitan() != null))
```
Comment (uniboi) [inline Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut:1113]: `else if((player.GetPetTitan())` might be a bit better to read

File: Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut [hunk 3]
```diff
@@ -1118,7 +1146,17 @@ void function PhaseEmbarkPhaseStop( entity player )
 	EmitSoundOnEntity( player, "pilot_phaseembark_end_3p" )
 	
 	if ( GetCurrentPlaylistVarInt( "fp_embark_enabled", 0 ) == 1 )
+	{
 		player.PhaseShiftCancel()
+	}
+	else if((player.GetPetTitan() != null))
```
Comment (uniboi) [inline Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut:1152]: `else if((player.GetPetTitan())` maybe

**Review by uniboi — COMMENTED** [review]
Some small things

**Comment by uniboi** [comment]
Are you still alive?

**Comment by Gityeeter** [comment]
I was busy with some other stuff with the past couple of months. Is there something i should do?

**Comment by x3Karma** [comment]
> I was busy with some other stuff with the past couple of months. Is there something i should do?

Honestly, not much, probably just check the suggested changes and maybe follow Northstar formatting if you're available.

File: Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut [hunk 4]
```diff
@@ -1014,11 +1023,20 @@ void function Embark_DelayedFadeOut( entity player, entity titan, float delay )
 	titan.EndSignal( "OnDestroy" )
 		
 	wait delay - EMBARK_FADE_TIME
-
+	entity soul = titan.GetTitanSoul()
 	if ( GetCurrentPlaylistVarInt( "fp_embark_enabled", 0 ) == 0 )
 	{
-		ScreenFadeToBlack( player, EMBARK_FADE_TIME, EMBARK_FADE_TIME + 0.2 ) // a little extra so we stay black
-		wait EMBARK_FADE_TIME
+		if ( !IsValid( soul.soul.armBadge ))
```
Comment (uniboi) [inline Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut:1029]:
```suggestion
		if ( !IsValid( soul.soul.armBadge ) )
```

File: Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut (hunk 2 above)
Comment (uniboi) [inline Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut:1113]:
```suggestion
	else if( IsValid( player.GetPetTitan() )
```

File: Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut [hunk 5]
```diff
@@ -1087,9 +1105,19 @@ void function PhaseEmbarkPhaseStart( entity player )
...
-	
+
 	if ( GetCurrentPlaylistVarInt( "fp_embark_enabled", 0 ) == 1 )
+	{
 		player.PhaseShiftBegin( 0.0, 0.2 )
+	}
+	else if((player.GetPetTitan() != null))
+	{
+		entity soul = player.GetPetTitan().GetTitanSoul()
+		if ( IsValid( soul.soul.armBadge ))
```
Comment (uniboi) [inline Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut:1116]:
```suggestion
		if ( IsValid( soul.soul.armBadge ) )
```

File: Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut (hunk 3 above)
Comment (uniboi) [inline Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut:1152]:
```suggestion
	else if( IsValid( player.GetPetTitan() )
```

File: Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut [hunk 6]
```diff
@@ -1118,7 +1146,17 @@ void function PhaseEmbarkPhaseStop( entity player )
 	EmitSoundOnEntity( player, "pilot_phaseembark_end_3p" )
 	
 	if ( GetCurrentPlaylistVarInt( "fp_embark_enabled", 0 ) == 1 )
+	{
 		player.PhaseShiftCancel()
+	}
+	else if((player.GetPetTitan() != null))
+	{
+		entity soul = player.GetPetTitan().GetTitanSoul()
+		if ( IsValid( soul.soul.armBadge ))
```
Comment (uniboi) [inline Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut:1155]:
```suggestion
		if ( IsValid( soul.soul.armBadge ) )
```

**Review by uniboi — CHANGES_REQUESTED** [review]
Just some more formatting, the rest looks good.

**Pull request closed without merging**